Sources/NIOSSL/SSLContext.swift

```diff
@@ -163,7 +163,27 @@ public final class NIOSSLContext {
         try NIOSSLContext.configureCertificateValidation(context: context,
                                                       verification: configuration.certificateVerification,
                                                       trustRoots: configuration.trustRoots)
-
+        
+        // Configure verification algorithms
+        if let verifySignatureAlgorithms = configuration.verifySignatureAlgorithms {
+            returnCode = verifySignatureAlgorithms
+                .map { $0.rawValue }
+                .withUnsafeBufferPointer { algo in
+                    CNIOBoringSSL_SSL_CTX_set_verify_algorithm_prefs(context, algo.baseAddress, algo.count)
+            }
+            precondition(1 == returnCode)
```

Comment:
It would be better if these preconditions were replaced with errors, ideally ones that used the BoringSSL error stack. `BoringSSL.unknownError` would be fine for this, along with `BoringSSLError.buildErrorStack()` to get the error stack.